rust_src/crates/webrender/src/window_system/winit/frame.rs

```diff
@@ -0,0 +1,346 @@
+use super::cursor::build_mouse_cursors;
+use super::cursor::emacs_to_winit_cursor;
+use crate::event_loop::WrEventLoop;
+use crate::output::Canvas;
+use crate::output::CanvasRef;
+use crate::window_system::api::monitor::MonitorHandle;
+use crate::window_system::output::OutputRef;
+use emacs::globals::Qfullscreen;
+use emacs::globals::Qmaximized;
+use emacs::{
+    bindings::{
+        fullscreen_type, list4i, make_frame, make_frame_without_minibuffer, make_minibuffer_frame,
+        output_method, winit_output, Emacs_Cursor,
+    },
+    frame::LispFrameRef,
+    globals::{Qinner_edges, Qnil, Qnone, Qonly, Qouter_edges},
+    keyboard::KeyboardRef,
+    lisp::LispObject,
+};
+use raw_window_handle::RawDisplayHandle;
+use raw_window_handle::RawWindowHandle;
+use webrender::api::ColorF;
+use webrender::{self, api::units::*};
+
+use super::output::Output;
+use crate::frame::LispFrameExt;
+
+use crate::window_system::api::{
+    dpi::{LogicalPosition, PhysicalPosition},
+    window::WindowBuilder,
+};
+
+use crate::display_info::DisplayInfoRef;
+
+pub type FrameID = crate::window_system::api::window::WindowId;
+
+pub trait LispFrameWinitExt {
+    fn build(
+        display: LispObject,
+        dpyinfo: DisplayInfoRef,
+        tem: LispObject,
+        kb: KeyboardRef,
+    ) -> Self;
+    fn set_window(&self, handle: crate::window_system::api::window::Window);
+    #[cfg(use_winit)]
+    fn set_cursor_position(&self, pos: PhysicalPosition<f64>);
+    fn set_visible2(&mut self, visible: bool);
```

Comment:
why set_visible2 instead of set_visible

Reply:
Oh, `set_visible` is taken by generated `bindings.rs` for `frame`.
Would you propose another name? I saw a lot naming like from emacs c files.

Reply:
Changed to `set_visible_`